<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsRelationManagerAddPolymorphicDialogBase</class>
 <widget class="QWidget" name="QgsRelationManagerAddPolymorphicDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout" columnstretch="1,2,0">
   <item row="2" column="0">
    <widget class="QLabel" name="mReferencedLayerFieldLayer">
     <property name="text">
      <string>Layer field</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="mIdLabel">
     <property name="text">
      <string>Id</string>
     </property>
    </widget>
   </item>
   <item row="8" column="1" colspan="2">
    <widget class="QgsCheckableComboBox" name="mReferencedLayersComboBox"/>
   </item>
   <item row="6" column="0" colspan="2">
    <widget class="QLabel" name="mFieldsMappingLabel">
     <property name="text">
      <string>Fields mapping</string>
     </property>
    </widget>
   </item>
   <item row="7" column="2">
    <layout class="QVBoxLayout" name="mFieldsMappingButtonsVBoxLayout">
     <item>
      <widget class="QToolButton" name="mFieldsMappingAddButton">
       <property name="toolTip">
        <string>Add new field pair as part of a composite foreign key</string>
       </property>
       <property name="icon">
        <iconset theme="add">
         <normaloff>../../../../../../.designer/backup</normaloff>../../../../../../.designer/backup</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="mFieldsMappingRemoveButton">
       <property name="toolTip">
        <string>Remove the select or last pair of fields</string>
       </property>
       <property name="icon">
        <iconset theme="remove">
         <normaloff>../../../../../../.designer/backup</normaloff>../../../../../../.designer/backup</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="mFieldsMappingVSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="mReferencingLayerLabel">
     <property name="text">
      <string>Referencing layer</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0" colspan="2">
    <widget class="QTableWidget" name="mFieldsMappingTable">
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="rowCount">
      <number>0</number>
     </property>
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>180</number>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <column>
      <property name="text">
       <string comment="polymorphic relations">Referenced (parent)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Referencing (child)</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="1" column="1" colspan="2">
    <widget class="QgsMapLayerComboBox" name="mReferencingLayerComboBox"/>
   </item>
   <item row="3" column="1" colspan="2">
    <widget class="QgsFieldExpressionWidget" name="mReferencedLayerExpressionWidget"/>
   </item>
   <item row="8" column="0">
    <widget class="QLabel" name="mReferencedLayersLabel">
     <property name="text">
      <string>Referenced layers</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="mReferencedLayerExpressionLabel">
     <property name="text">
      <string>Layer field expression</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1" colspan="2">
    <widget class="QgsFieldComboBox" name="mReferencedLayerFieldComboBox"/>
   </item>
   <item row="0" column="1" colspan="2">
    <widget class="QLineEdit" name="mIdLineEdit">
     <property name="placeholderText">
      <string>[Generated automatically]</string>
     </property>
    </widget>
   </item>
   <item row="9" column="0" colspan="3">
    <widget class="QLabel" name="mPolymorphicRelationHelpLabel">
     <property name="text">
      <string>Polymorphic relations are for advanced usage where a set of standard relations have the same referencing layer but the referenced layer is calculated from an expression.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="10" column="0" colspan="3">
    <widget class="QDialogButtonBox" name="mButtonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Help|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsCheckableComboBox</class>
   <extends>QComboBox</extends>
   <header>qgscheckablecombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsFieldComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsfieldcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsFieldExpressionWidget</class>
   <extends>QWidget</extends>
   <header>qgsfieldexpressionwidget.h</header>
  </customwidget>
  <customwidget>
   <class>QgsMapLayerComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsmaplayercombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
